src/test/java/uk/gov/companieshouse/limitedpartnershipsapi/service/LimitedPartnershipServiceTest.java

```diff
@@ -61,15 +64,16 @@ void givenDto_whenCreateLP_thenLPCreatedWithSubmissionIdAndTransactionUpdated()
         when(mapper.dtoToDao(limitedPartnershipSubmissionDto)).thenReturn(limitedPartnershipSubmissionDao);
         when(repository.insert(limitedPartnershipSubmissionDao)).thenReturn(limitedPartnershipSubmissionDao);
 
-        Transaction transaction = buildTransaction();
-
         // when
-        String submissionId = service.createLimitedPartnership(transaction, limitedPartnershipSubmissionDto, REQUEST_ID, USER_ID);
+        String submissionId = service.createLimitedPartnership(buildTransaction(),limitedPartnershipSubmissionDto, REQUEST_ID, USER_ID);
+        Transaction transaction = buildTransaction();
```

Comment:
This needs to revert back to the original code where `buildTransaction()` is only invoked once.

Reply:
reverted back to one invocation
```suggestion
        Transaction transaction = buildTransaction();
```

Reply:
This should move out of the `// when` block and back up a few lines, to where it was originally (it's part of the test fixture and not the method/code under test).

Reply:
original line returned